src/main/java/org/prebid/server/bidder/sharethrough/SharethroughMarkupUtil.java

```diff
@@ -0,0 +1,58 @@
+package org.prebid.server.bidder.sharethrough;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.vertx.core.json.Json;
+import org.apache.commons.lang3.BooleanUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.prebid.server.bidder.sharethrough.model.StrUriParameters;
+import org.prebid.server.bidder.sharethrough.model.bidResponse.ExtImpSharethroughResponse;
+
+import java.util.Base64;
+
+class SharethroughMarkupUtil {
+
+    private SharethroughMarkupUtil() {
+    }
+
+    static String getAdMarkup(ExtImpSharethroughResponse strResponse, StrUriParameters strUriParameters)
+            throws JsonProcessingException {
+        final StringBuilder tmplBody = new StringBuilder();
+
+        final String strRespId = String.format("str_response_%s", strResponse.getBidId());
+        final String arId = strResponse.getAdserverRequestId();
+        final String jsonStrResponse = Json.mapper.writeValueAsString(strResponse);
+        if (StringUtils.isBlank(jsonStrResponse)) {
+            return "";
```

Comment:
Please check the Go version - they return empty string `""` just to return something, while returning an error in the first place to catch it above. I think this check is unnecessary since all they do in Go is just a check if any error occurred during writing to json string. Please remove this `if` block.